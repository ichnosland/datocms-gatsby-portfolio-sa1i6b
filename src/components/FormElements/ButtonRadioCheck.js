/**
*
* ButtonRadioCheck
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { colore } from 'style/color';
import { RadioCheckInput } from './InputRadioCheck';

export const ButtonRadioCheckLabel = styled.label`
  position: relative;
  display: block;
  padding: 12px;
  margin: 10px 0;
  background-color: ${colore.ui.contrast};
  border: 1px solid ${(props) => props.theme.brand};
  border-bottom-width: 3px;
  border-radius: ${(props) => props.theme.radius.general};
  ${({ checked }) => checked && css`
    color: ${(props) => props.theme.brand};
    border-width: 1px;
    &:after{
      content: '';
      display: block;
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      border-radius: 5px;
      border: 3px solid ${(props) => props.theme.pale};
    }
  `};
`;

ButtonRadioCheckLabel.defaultProps = {
  theme: {
    brand: '#00BBEF',
    pale: 'rgb(199, 231, 245)',
    radius: {
      general: '6px',
    },
  },
};

class ButtonRadioCheck extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { label, type, id, input } = this.props;
    return (
      <ButtonRadioCheckLabel
        type={type}
        checked={input.checked}
        htmlFor={id}
      >
        <RadioCheckInput
          {...input}
          type={type}
          id={id}
        />
        {label}
      </ButtonRadioCheckLabel>
    );
  }
}

ButtonRadioCheck.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string.isRequired,
  id: PropTypes.string,
  input: PropTypes.object,
};

export default ButtonRadioCheck;
