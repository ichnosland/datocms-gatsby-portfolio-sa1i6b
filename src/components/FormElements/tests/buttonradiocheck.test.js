import React from 'react';
import { shallow } from 'enzyme';

import ButtonRadioCheck, { ButtonRadioCheckLabel } from '../ButtonRadioCheck';

const type = 'radio';
const input = {
  checked: true,
};

describe('<ButtonRadioCheck />', () => {
  it('should render a ButtonRadioCheckLabel', () => {
    const renderedComponent = shallow(<ButtonRadioCheck type={type} input={input} />);
    expect(renderedComponent.find(ButtonRadioCheckLabel).length).toEqual(1);
  });

  it('should render its css w/ props type', () => {
    const renderedComponent = shallow(<ButtonRadioCheck type={type} input={input} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its label w/ props label', () => {
    const label = 'Etichetta';
    const renderedComponent = shallow(<ButtonRadioCheck type={type} input={input} label={label} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});


describe('<ButtonRadioCheckLabel />', () => {
  it('should render a <label> tag', () => {
    const renderedComponent = shallow(<ButtonRadioCheckLabel />);
    expect(renderedComponent.type()).toEqual('label');
  });

  it('should render its css w/ props checked', () => {
    const checked = true;
    const renderedComponent = shallow(<ButtonRadioCheckLabel checked={checked} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ theme', () => {
    const theme = {
      brand: '#00BBEF',
      pale: 'rgb(199, 231, 245)',
      radius: {
        general: '6px',
      },
    };
    const renderedComponent = shallow(<ButtonRadioCheckLabel type={type} theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
