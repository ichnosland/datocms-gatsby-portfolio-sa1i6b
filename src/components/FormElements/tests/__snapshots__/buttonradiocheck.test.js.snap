// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonRadioCheck /> should render its css w/ props type 1`] = `
<ButtonRadioCheck__ButtonRadioCheckLabel
  checked={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "pale": "rgb(199, 231, 245)",
      "radius": Object {
        "general": "6px",
      },
    }
  }
  type="radio"
>
  <InputRadioCheck__RadioCheckInput
    checked={true}
    type="radio"
  />
</ButtonRadioCheck__ButtonRadioCheckLabel>
`;

exports[`<ButtonRadioCheck /> should render its label w/ props label 1`] = `
<ButtonRadioCheck__ButtonRadioCheckLabel
  checked={true}
  theme={
    Object {
      "brand": "#00BBEF",
      "pale": "rgb(199, 231, 245)",
      "radius": Object {
        "general": "6px",
      },
    }
  }
  type="radio"
>
  <InputRadioCheck__RadioCheckInput
    checked={true}
    type="radio"
  />
  Etichetta
</ButtonRadioCheck__ButtonRadioCheckLabel>
`;

exports[`<ButtonRadioCheckLabel /> should render its css w/ props checked 1`] = `
.c0 {
  position: relative;
  display: block;
  padding: 12px;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border: 1px solid #00BBEF;
  border-bottom-width: 3px;
  border-radius: 6px;
  color: #00BBEF;
  border-width: 1px;
}

.c0:after {
  content: '';
  display: block;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 5px;
  border: 3px solid rgb(199,231,245);
}

<label
  checked={true}
  className="c0"
/>
`;

exports[`<ButtonRadioCheckLabel /> should render its css w/ theme 1`] = `
.c0 {
  position: relative;
  display: block;
  padding: 12px;
  margin: 10px 0;
  background-color: rgb(255,255,255);
  border: 1px solid #00BBEF;
  border-bottom-width: 3px;
  border-radius: 6px;
}

<label
  className="c0"
  type="radio"
/>
`;
